src/FileDownloader.ts

```diff
@@ -59,24 +57,56 @@ export default class FileDownloader implements IFileDownloader {
         const retries = settings?.retries ?? DefaultRetries;
         const retryDelayInMs = settings?.retryDelayInMs ?? DefaultRetryDelayInMs;
 
-        const downloadStream: Readable = await this._requestHandler.get(
-            url.toString(),
-            timeoutInMs,
-            retries,
-            retryDelayInMs,
-            cancellationToken,
-            onDownloadProgressChange
-        );
-
-        const writeStream: Writable = settings?.shouldUnzip ?? false
-            ? ZipDecompressor(tempFileDownloadPath)
-            : fs.createWriteStream(tempFileDownloadPath);
-
-        // Start the download and wait for it to completely finish writing to disk
-        // The first element in the pipeline must be the readableStream and the last element must be the writableStream
-        const pipelinePromise = pipelineAsync([downloadStream, writeStream]);
-        const writeStreamClosePromise = new Promise(resolve => writeStream.on(`close`, resolve));
-        await Promise.all([pipelinePromise, writeStreamClosePromise]);
+        // Everything inside this try is a dirty fix to try to make things more reliable. A better fix using piped streams properly is needed.
+        try{
+            // Fake report progress code while we wait for the real fix
+            let progress = 0;
+            const progressTimerId = setInterval(() => {
+                if (progress <= 100) {
+                    // TODO: the whole timer should be under this if.
+                    if(onDownloadProgressChange != null){
+                        onDownloadProgressChange(progress++, 100);
```

Comment:
we expect this to be bytes downloaded, and calculate percentages on our extension. Is this changing that behaviour and should we change our extension code as well?

Reply:
So this is a fake progress reporter. To make things easier instead of reporting 20.000.000 bytes out of 100.000.000 I just report 20 out of 100.